pallets/xcm-app/src/benchmarking.rs

```diff
@@ -31,104 +31,89 @@
 use super::*;
 use crate::Pallet as XCMApp;
 use frame_benchmarking::{benchmarks, impl_benchmark_test_suite};
-use frame_support::WeakBoundedVec;
 use frame_system::RawOrigin;
-use xcm::{
-    opaque::latest::{
-        Junction::{GeneralKey, Parachain},
-        Junctions::X2,
-    },
-    v1::MultiLocation,
-};
+use xcm::{opaque::latest::Junction::GeneralKey, v3::MultiLocation};
+
+fn alice<T: Config>() -> T::AccountId {
+    let bytes = [66; 32];
+    T::AccountId::decode(&mut &bytes[..]).expect("Failed to decode account ID")
+}
 
 benchmarks! {
-    register_mapping {
+    register_asset {
         let asset_id = [1; 32].into();
-        let multilocation = MultiLocation {
-            parents: 1,
-            interior: X2(Parachain(666), GeneralKey(test_general_key())),
-        };
-    }: _(RawOrigin::Root, asset_id, multilocation.clone())
+        let multilocation = test_multilocation();
+    }: _(RawOrigin::Root, asset_id, multilocation.clone().into(), bridge_types::types::AssetKind::Thischain, 1000)
```

Comment:
You should use `T::CallOrigin::try_successful_origin` instead of `RawOrigin::Root` here and everywhere where `T::CallOrigin::ensure_origin` is used

For example https://github.com/sora-xor/sora2-common/blob/bb77d5beeabc3ec056b2cd7eeb750887683084ee/pallets/data-signer/src/benchmarking.rs#L115

Reply:
done